### PR TITLE
Deprecation: Deprecate kwargs in CausalGraph construction

### PR DESCRIPTION
**Pull request by maxelliott-causalens**
## Description and Motivation

- Passing `input_list`, `output_list`, and `fully_connected` as arguments to the constructor of a
  `cai_causal_graph.causal_graph.CausalGraph` is now deprecated, and will be removed in future versions. Please add
  nodes and edges after construction instead.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [x] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [x] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [ ] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [x] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

File: cai_causal_graph/causal_graph.py [hunk 1]
```diff
@@ -473,12 +474,27 @@ def __init__(
 
         # Add nodes.
         if input_list is not None:
+            warnings.warn(
+                'Passing input_list during construction of a CausalGraph is now deprecated, and will be '
+                'removed in future versions. Please add nodes and edges after construction instead.',
```
Comment (steven-causalens) [inline cai_causal_graph/causal_graph.py:479]: Would suggest to change "Please add nodes and edges after construction instead." to "Please add nodes via the add_node method and edges via the add_edge method instead."

Just to be a bit more specific. What do you think?

File: cai_causal_graph/causal_graph.py [hunk 2]
```diff
@@ -473,12 +474,27 @@ def __init__(
...
+                category=DeprecationWarning,
+            )
             self.add_nodes_from(input_list)
         if output_list is not None:
+            warnings.warn(
+                'Passing output_list during construction of a CausalGraph is now deprecated, and will be '
+                'removed in future versions. Please add nodes and edges after construction instead.',
```
Comment (steven-causalens) [inline cai_causal_graph/causal_graph.py:486]: Same here

File: cai_causal_graph/causal_graph.py [hunk 3]
```diff
@@ -473,12 +474,27 @@ def __init__(
...
+                category=DeprecationWarning,
+            )
             self.add_nodes_from(output_list)
 
         # create a fully-connected causal graph if specified, using only directed edges
         if fully_connected and input_list is not None and output_list is not None:
+            warnings.warn(
+                'Passing fully_connected during construction of a CausalGraph is now deprecated, and will be '
+                'removed in future versions. Please add nodes and edges after construction instead.',
```
Comment (steven-causalens) [inline cai_causal_graph/causal_graph.py:495]: Same here

**Review by steven-causalens — APPROVED** [review]
Thanks, just one comment on the warning message

File: cai_causal_graph/causal_graph.py (hunk 1 above)
Comment (maxelliott-causalens) [reply cai_causal_graph/causal_graph.py]: Resolved offline, pointing to add_fully_connected_nodes instead.

**Review by steven-causalens — APPROVED** [review]
LGTM 🙏

**Pull request merged**